rfcs/0003-storage.md

```diff
@@ -1,46 +1,108 @@
+8. Storage mining capacity: the number of keys and samples that a storage miner can verify each storage epoch.
+
+## 6.2 Constants
+
+1. NUM\_STORAGE\_ENTRIES: Number of entries in a block of ledger data. The unit of storage for a replicator.
+3. NUM\_KEY\_ROTATION\_TICKS: Number of ticks to save a PoH value and cause a key generation for the section of ledger just generated and the rotation of another key in the set.
+4. NUM\_STORAGE\_PROOFS: Number of storage proofs required for a storage proof claim to be successfully rewarded.
+5. RATIO\_OF\_FAKE\_PROOFS: Ratio of fake proofs to real proofs that a storage mining proof claim has to contain to be valid for a reward.
+6. NUM\_STORAGE\_SAMPLES: Number of samples required for a storage mining proof.
+7. NUM\_CHACHA\_ROUNDS: Number of encryption rounds performed to generate encrypted state.
+
+## 6.3 Validator behavior
+
+1. Validator joins the network and submits a storage mining capacity transaction which tells the network how many proofs it can process in a given period defined by NUM\_KEY\_ROTATION\_TICKS.
+2. Every NUM\_KEY\_ROTATION\_TICKS the validator stores the PoH value at that height.
+4. Every NUM\_KEY\_ROTATION\_TICKS it also validates samples received from replicators. It signs the PoH hash at that point and uses the low 5 bits of the first byte to index into the signature and then picks a number of bytes of the received sha states with that one in choosing which storage proofs to validate. It then creates a mask of valid proofs and fake proofs and sends it to the leader. This is a storage proof confirmation transaction.
+5. The storage proof confirmation transaction is integrated into the ledger.
+6. After a lockout period of NUM\_SECONDS\_STORAGE\_LOCKOUT seconds, the validator then submits a storage proof claim transaction which then causes the distribution
+of the storage reward if no challenges were seen for the proof to the validators and replicators party to the proofs.
+7. Validator responds to RPC interfaces for what its encryption key state is and PoH values for any rotation events.
+
+## 6.4 Replicator behavior
+
+1. Replicator paranoia modes:
+For many of the operations there are number of options depending on how paranoid a replicator is:
+- (a) replicator can ask a validator
+- (b) replicator can ask multiple validators
+- (c) replicator can subscribe to the full transaction stream and generate the key state itself
+- (d) replicator can subscribe to an abbreviated transaction stream to generate the key state itself
+2. A replicator obtains the PoH hash corresponding to the last key rotation.
+3. The replicator signs the PoH hash with its keypair. That signature is the seed used to pick the block to replicate and also the encryption key.
+4. The replicator sends a request to a validator with its signature asking for which block it should now store. The validator will respond with the starting entry height and the number of entries.
```

Comment:
can't it compute that on its own?

Reply:
Well I was thinking the replicator wouldn't know even the max ledger entry height at this point, maybe it could ask that. So if does that, then it could use some bits in the signature to choose which block?

Reply:
Also limits the network to a fixed block size.

Reply:
@sakridge the last id that it signs to generate the seed would define the max height.  it can't be higher then the last id.